src/test/java/ParserTest.java

```diff
@@ -0,0 +1,80 @@
+import duke.DukeException;
+import duke.Parser;
+import duke.TaskList;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class ParserTest {
+    @Test
+    public void write_ToDoCommand_writtenCorrectly() {
+        Parser testParser = new Parser(new TaskList());
+        String testToDoStr = "todo return book to library";
+
+        try {
+            assertEquals(testParser.parse(testToDoStr), "Got it. I've added this task:\n  " +
+                    "[T][✘] return book to library" + "\nNow you have 1 tasks in the list.");
+            System.out.println("Todo Command correctly parsed.");
+        } catch (DukeException e) {
+            System.out.println(e.getMessage());
+            fail();
+        }
+    }
+
+    @Test
+    public void write_ToDoCommand_empty_exceptionThrown() {
+        Parser testParser = new Parser(new TaskList());
+        String testToDoStr = "todo ";
+
+        try {
+            assertEquals(testParser.parse(testToDoStr), "Got it. I've added this task:\n  " +
+                    "[T][✘] return book to library" + "\nNow you have 1 tasks in the list.");
+        } catch (DukeException e ) {
+            System.out.println("Exception successfully thrown: " + e.getMessage());
+        }
+    }
+
+    @Test
+    public void write_deadLineCommand_writtenCorrectly() {
+        Parser testParser = new Parser(new TaskList());
+        String testToDoStr = "deadline return book /by 12/12/1212 1212";
+
+        try {
+            assertEquals(testParser.parse(testToDoStr), "Got it. I've added this task:\n  " +
+                    "[D][✘] return book (by: 12/12/1212 1212)" + "\nNow you have 1 tasks in the list.");
+            System.out.println("deadline Command correctly parsed.");
+        } catch (DukeException e) {
+            System.out.println(e.getMessage());
+            fail();
+        }
+    }
+
+    @Test
+    public void write_deadLineCommand_empty_exceptionThrown() {
+        Parser testParser = new Parser(new TaskList());
+        String testToDoStr = "deadline ";
+
+        try {
+            assertEquals(testParser.parse(testToDoStr), "Got it. I've added this task:\n  " +
+                    "[D][✘] return book (by: 12/12/1212 1212)" + "\nNow you have 1 tasks in the list.");
```

Comment:
Just wondering, how were you able to get the console to print the icon? As I have errors printing icon accurately. Thank you!

Reply:
Hmm, I am unsure, I just used the unicodes as was given, namely the "\u2713" and "\u2718" for the tick or X symbols respectively. Were you able to initially print the icons for the classes first? Or was this only a problem upon unit testing?